lambda/lambda.ml

```diff
@@ -212,7 +212,8 @@ type structured_constant =
   | Const_immstring of string
 
 type tailcall_attribute =
-  | Should_be_tailcall (* [@tailcall] *)
+  | Should_be_tailcall (* [@tailcall] or [@tailcall true] *)
+  | Should_not_be_tailcall (* [@tailcall false] *)
```

Comment:
Perhaps these two constructors could be merged into a single one `Expect_tailcall of bool`? It would match up with the warning constructor and the call to `maybe_warn` below...

Reply:
Initially I didn't want to do this because I think that `bool` is not clear enough on what it means. (Does `false` mean that the implementation should try to disable tail-call optimization when we are in tailcall position?) But with `Expect` in the name of some form, it is sort of clear that the attribute semantics is (currently) only to report an expectation, so it is in fact equivalent to the current names, but more factorized. Will change.